### PR TITLE
Unify `TestBatchNormalization` and `TestBatchNormalizationAxis`

### PR DESCRIPTION
**Pull request by toslunar**
Add ideep tests to `TestBatchNormalizationAxis`.  Then unify the code with `TestBatchNormalization`.

#4266

### CODE REVIEW

File: tests/chainer_tests/functions_tests/normalization_tests/test_batch_normalization.py [hunk 1]
```diff
@@ -53,176 +63,47 @@ def _batch_normalization(args):
...
-        param_shape = self.param_shape
         dtype = self.dtype
-        ndim = self.ndim
+        aggr_axes = self.axis
+
+        if aggr_axes is None:
+            param_shape = self.param_shape
+            ndim = self.ndim
+            shape = (5,) + param_shape + (2,) * ndim
+
```
Comment (delta2323) [inline tests/chainer_tests/functions_tests/normalization_tests/test_batch_normalization.py]: Remove this empty line.

**Comment by delta2323** [comment]
jenkins, test this please.

**Comment by delta2323** [comment]
The CI failed due to a warning that is seeming irrelevant to this PR. Could you merge or rebase the latest master branch?

**Comment by toslunar** [comment]
Done

**Comment by delta2323** [comment]
Thank you!

**Comment by delta2323** [comment]
jenkins, test this please.

**Comment by delta2323** [comment]
jenkins, test this please.

**Comment by delta2323** [comment]
LGTM

**Comment by delta2323** [comment]
LGTM

**Pull request merged**